Tests/src/PublicAPITests/MaterialTests.cpp

```diff
@@ -0,0 +1,330 @@
+
+	// Cleanup
+	DeleteMaterial(AssetSystem, CreatedMaterial);
+
+	DeleteSpace(SpaceSystem, Space.Id);
+	LogOut(UserSystem);
+}
+#endif
+
+#if RUN_ALL_UNIT_TESTS || RUN_MATERIAL_TESTS || RUN_MATERIAL_GETEMPTYMATERIALS_TEST
+
+CSP_PUBLIC_TEST(CSPEngine, MaterialTests, GetEmptyMaterialsTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = ::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* AssetSystem	 = SystemsManager.GetAssetSystem();
+
+	// Log in
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	// Create space to search for materials
+	::Space Space;
+	CreateDefaultTestSpace(SpaceSystem, Space);
+
+	// Attempt to find materials in a space none have been created for
+	csp::common::Array<GLTFMaterial> FoundMaterials;
+	GetMaterials(AssetSystem, Space.Id, FoundMaterials);
+
+	EXPECT_TRUE(FoundMaterials.IsEmpty());
+
+	// Cleanup
+	DeleteSpace(SpaceSystem, Space.Id);
+	LogOut(UserSystem);
+}
+#endif
+
+#if RUN_ALL_UNIT_TESTS || RUN_MATERIAL_TESTS || RUN_MATERIAL_GETMULTIPLEMATERIALS_TEST
+
+CSP_PUBLIC_TEST(CSPEngine, MaterialTests, GetMultipleMaterialsTest)
+{
+	SetRandSeed();
+
+	auto& SystemsManager = ::SystemsManager::Get();
+	auto* UserSystem	 = SystemsManager.GetUserSystem();
+	auto* SpaceSystem	 = SystemsManager.GetSpaceSystem();
+	auto* AssetSystem	 = SystemsManager.GetAssetSystem();
+
+	// Log in
+	csp::common::String UserId;
+	LogIn(UserSystem, UserId);
+
+	// Create space to search for materials
+	::Space Space;
+	CreateDefaultTestSpace(SpaceSystem, Space);
+
+	// Create 3 materials
+	constexpr const char* TestMaterialName1 = "TestMaterial1";
+	GLTFMaterial CreatedMaterial1;
+	CreateMaterial(AssetSystem, TestMaterialName1, Space.Id, CreatedMaterial1);
+
+	constexpr const char* TestMaterialName2 = "TestMaterial2";
+	GLTFMaterial CreatedMaterial2;
+	CreateMaterial(AssetSystem, TestMaterialName2, Space.Id, CreatedMaterial2);
+
+	constexpr const char* TestMaterialName3 = "TestMaterial3";
+	GLTFMaterial CreatedMaterial3;
+	CreateMaterial(AssetSystem, TestMaterialName3, Space.Id, CreatedMaterial3);
+
+	// Attempt to find the 3 materials that have been created
+	csp::common::Array<GLTFMaterial> FoundMaterials;
+	GetMaterials(AssetSystem, Space.Id, FoundMaterials);
+
+	EXPECT_EQ(FoundMaterials.Size(), 3);
+
+	// Ensure we found the right materials
+	std::vector<csp::common::String> MaterialNames {TestMaterialName1, TestMaterialName2, TestMaterialName3};
+
+	for (int i = 0; i < FoundMaterials.Size(); ++i)
+	{
+		const csp::common::String& SearchName = FoundMaterials[i].GetName();
+
+		auto Found = std::find_if(std::begin(MaterialNames),
+								  std::end(MaterialNames),
+								  [&SearchName](const csp::common::String& Name)
+								  {
+									  return Name == SearchName;
+								  });
+
+
+		EXPECT_TRUE(Found != MaterialNames.end());
+	}
+
+	// Cleanup
+	DeleteMaterial(AssetSystem, CreatedMaterial1);
```

Comment:
[Question] would it add any value to the test to first delete one, say number 2, and check that the other two (1 and 3) are still there, and only then remove 1 and 3?

Reply:
I can add a separate test for deletion

Reply:
done